### PR TITLE
refactor project and implement Create a new account inside an opportu…

### PR DESCRIPTION
**Pull request by oscarlopez8**
…nity

### CODE REVIEW

File: src/main/java/salesforce/api/rest/RestClientAPI.java [hunk 1]
```diff
@@ -109,7 +109,10 @@ private void requestAuthentication() {
      */
     public Response post(final String endpoint, final Map<String, String> valuesForTheBody) {
         url = urlAPI.concat(endpoint);
+
         response = request.body(valuesForTheBody).when().post(url);
+        int status = response.statusCode();
+        System.out.println(status);
```
Comment (LimbertVargas) [inline src/main/java/salesforce/api/rest/RestClientAPI.java:115]: please check the extra line and delete the Sout

File: src/main/java/salesforce/entities/Opportunity.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    public String getAmount() {
+        return amount;
+    }
+
+    /**
+     * Sets the id of Opportunity sending a string.
+     *
+     * @param amount for the Opportunity.
+     */
+    public void setAmount(final String amount) {
```
Comment (LimbertVargas) [inline src/main/java/salesforce/entities/Opportunity.java]: Why your variable name is "amount"? the variable name should be more descriptive

File: src/main/java/salesforce/ui/pages/TransporterPage.java [hunk 3]
```diff
@@ -73,6 +69,7 @@ public static void navigateToURL(final String uriPage) {
      * @return a home page according to the page layout.
      */
     public HomePage navigateToHomePage() {
+
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/TransporterPage.java]: please delete this line

File: src/main/java/salesforce/ui/pages/cases/CaseClassicForm.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * Waits until a webElement is loaded.
+     */
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
+
+    }
+}
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/cases/CaseClassicForm.java]: please add code in the methods of this class

File: src/main/java/salesforce/ui/pages/cases/CaseClassicPage.java [hunk 5]
```diff
@@ -35,19 +35,28 @@
      *
      * @return an instance web element.
      */
+
+
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/cases/CaseClassicPage.java]: delete extra line

File: src/main/java/salesforce/ui/pages/cases/CaseClassicPage.java [hunk 6]
```diff
@@ -35,19 +35,28 @@
...
+    /**
+     * Returns an Case Form page after clicking on new button.
+     *
+     * @return an case form page.
+     */
     @Override
-    public WebElement buttonNew() {
-        return null;
+    public CaseFormAbstract clickNewBtn() {
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/cases/CaseClassicPage.java:39]: add documentation for this method

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    @Override
+    public String getAccountName() {
+        return accountName.getAttribute("value");
+    }
+
+
+    @Override
+    public void goToFormPageAbstract() {
+        // To do
+    }
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java:146]: complete this method

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    @FindBy(css = "span[title='New Account']")
+    private WebElement newAccount;
+
+    @FindBy(css = "div[title='John Doe Test']")
+    private WebElement nameAccount;
+
+    @FindBy(css = "span[class='pillText']")
+    private WebElement pillNameAccount;
+
+
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityLightningPage.java]: delete this extra line

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityPageAbstract.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Click On Opportunity.
+     * @param idOpportunity param.
+     */
+    public abstract void clickOnOpportunity(String idOpportunity);
+
+    /**
+     * Returns a list with name Opportunity from the Opportunity page.
+     *
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityPageAbstract.java:35]: delete this extra line

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityPageAbstract.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Set Account Name.
+     * @param nameAccount param.
+     */
+    public abstract void setAccountName(String nameAccount);
+
+    /**
+     * Get Account Name.
+     *
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityPageAbstract.java:48]: delete this line

File: src/main/java/salesforce/ui/utils/SwitchWindows.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+package salesforce.ui.utils;
+
+import core.selenium.WebDriverManager;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.Set;
+
+/**
+ * Switch hWindows.
```
Comment (LimbertVargas) [inline src/main/java/salesforce/ui/utils/SwitchWindows.java]: improve this comment

File: src/test/java/steps/OpportunitySteps.java [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    private Context context;
+
+    /**
+     * Variable for the Opportunity Abstract page.
+     */
+    private OpportunityPageAbstract opportunityPage;
+
+    /**
+     *
```
Comment (LimbertVargas) [inline src/test/java/steps/OpportunitySteps.java]: add documentation

**Review by LimbertVargas — COMMENTED** [review]
please review comments

File: src/main/java/salesforce/entities/Opportunity.java (hunk 2 above)
Comment (oscarlopez8) [reply src/main/java/salesforce/entities/Opportunity.java]: In this line "* @param amount for the Opportunity." I describe the variable

**Review by LimbertVargas — APPROVED** [review]

**Pull request merged**